osu.Game/Rulesets/Mods/IApplicableToDrawableHitObject.cs

```diff
@@ -1,21 +1,30 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using System.Collections.Generic;
+using osu.Framework.Extensions.IEnumerableExtensions;
 using osu.Game.Rulesets.Objects.Drawables;
 
 namespace osu.Game.Rulesets.Mods
 {
     /// <summary>
     /// An interface for <see cref="Mod"/>s that can be applied to <see cref="DrawableHitObject"/>s.
     /// </summary>
-    public interface IApplicableToDrawableHitObjects : IApplicableMod
+    public interface IApplicableToDrawableHitObject : IApplicableMod
     {
         /// <summary>
-        /// Applies this <see cref="IApplicableToDrawableHitObjects"/> to a list of <see cref="DrawableHitObject"/>s.
+        /// Applies this <see cref="IApplicableToDrawableHitObject"/> to a <see cref="DrawableHitObject"/>.
         /// This will only be invoked with top-level <see cref="DrawableHitObject"/>s. Access <see cref="DrawableHitObject.NestedHitObjects"/> if adjusting nested objects is necessary.
         /// </summary>
-        /// <param name="drawables">The list of <see cref="DrawableHitObject"/>s to apply to.</param>
+        void ApplyToDrawableHitObject(DrawableHitObject drawable);
+    }
+
+    [Obsolete(@"Use the singular version IApplicableToDrawableHitObject instead.")] // Can be removed 20211216
```

Comment:
Not sure about deprecating this with perspective of removal. While it *is* true that all current `IApplicableToDrawableHitObjects` mods (including custom rulesets) are boilerplate-y as they perform adjustments individual to each object, it may still be an interesting API surface to expose if you wanted to i.e. apply an adjustment to only *some* objects, or somehow in a manner dependent on the previous object(s) without resorting to storing state in the mod class.

Reply:
I don't think such interface happens because it is incompatible to the hit object pooling.

Reply:
Yeah that's a fair point. On the other side that means that this sort of thing cannot be easily done without modifying the beatmap itself in some capacity. I guess maybe not worth worrying about for now until a use case actually happens.

I was only doubtful about removing as I thought it was a free capability that there was no big reason to remove, but that clearly isn't the case when pooling is active as it makes it behave in an unintuitive manner.

Reply:
At the point of renaming the interface, we're already breaking the API so I'm not sure we need this here.

If we're going to provide an obsolescence path, there should probably still be an `IApplableToDrawableHitObjects` interface, rather than adding the method here. If that's going to impede further code cleanups or fixes for pooling issues then I'd almost say it's not worth it.

Probably need @smoogipoo's thoughts on this as well.

Reply:
This isn't breaking the API as it is. I've also confirmed this with sentakki's AutoTouch mod.

Reply:
How...? Does it somehow magically handle interface naming changes or is there some fallback somewhere that I am missing?

Reply:
The old (plural) interface implements the new (singular), and has a default implementation of the plural that calls the plural with an array with a single DHO.